silica/transformations/desugar_for_loops.py

```diff
@@ -1,88 +1,157 @@
-            width = None
-            if isinstance(stop, ast.Num):
-                width = (stop.n - 1).bit_length()
-            else:
-                width = None
-                for keyword in node.iter.keywords:
-                    if keyword.arg == "bit_width":
-                        assert isinstance(keyword.value, ast.Num)
-                        width = keyword.value.n
-                        break
-                assert width is not None
-            self.loopvars.add((node.target.id, width))
-            return [
-                ast.Assign([ast.Name(node.target.id, ast.Store())], ast.parse(f"bits({start}, {width})").body[0].value),
-                ast.Assign([ast.Name(node.target.id + "_cout", ast.Store())], ast.NameConstant(False)),
-                ast.While(ast.parse(f"not_({node.target.id}_cout)").body[0].value,
-                    node.body + [
-                        ast.parse(f"{node.target.id}, {node.target.id}_cout = add({node.target.id}, bits({incr}, {width}), cout=True)").body[0]
-                    ], [])
-            ]
-        else:  # pragma: no cover
-            print_ast(node)
-            raise NotImplementedError("Unsupported for loop construct `{}`".format(to_source(node.iter)))
-
-def desugar_for_loops(tree):
+                    raise NotImplementedError("keyword iterators are not yet supported")
+
+                bit_width = eval(astor.to_source(stop).rstrip() + "-" + astor.to_source(start).rstrip()).bit_length()
+                self.loopvars.add((node.target.id, bit_width))
+
+                return [
+                    ast.Assign([ast.Name(node.target.id, ast.Store())], start),
+                    ast.While(ast.BinOp(ast.Name(node.target.id, ast.Load()), ast.Lt(), stop),
+                        node.body + [
+                            ast.Assign([ast.Name(node.target.id, ast.Store())], ast.BinOp(
+                                ast.Name(node.target.id, ast.Load()), ast.Add(), step))
+                        ], [])
+                ]
+            # for _ in name
+            elif is_name(node.iter):
+                # TODO: currently just assumes that this is a list
+                # TODO: maybe this should just reshape the for-loop to use a range()
+                #       iterator and then just call the previous branch
+                index = '__silica_count_' + str(next(counter))
+                start = ast.Num(0)
+                stop  = ast.Num(type_table[node.iter.id][1])
+                step  = ast.Num(1)
+
+                bit_width = stop.n.bit_length()
+                self.loopvars.add((index, bit_width))
+
+                return [
+                    ast.Assign([ast.Name(index, ast.Store())], start),
+                    ast.While(ast.BinOp(ast.Name(index, ast.Load()), ast.Lt(), stop),
+                        [ast.Assign([ast.Name(node.target.id, ast.Store())],
+                                    ast.Subscript(node.iter, ast.Index(ast.Name(index, ast.Load()))))] +
```

Comment:
It looks like travis is unhappy about this `Subscript` constructor call. Are you using Python 3.6? The AST has had some changes in Python 3, so that could explain the discrepancy if you're not seeing the issue locally.  Subscript expects `value, slice, ctx` (see https://greentreesnakes.readthedocs.io/en/latest/nodes.html#Subscript), so I think this is missing a `ctx`. I think in this case it would be an `ast.Load()`, since we're loading the subscripted value and storing into the loop var